### ISSUE TITLE
Unable to parse numeric value '.' to float64 when using a struct parameter.

### ISSUE THREAD

**Issue by kennethkoontz** [Knetic/govaluate#61]
When passing in an expression that contains a struct to `NewEvaluableExpression`, I'm getting "Unable to parse numeric value '.' to float64" error.

```go

// element.ShowCondition.Condition = "Survey.Element('q1').Row('r2').Selected()"
expression, err := govaluate.NewEvaluableExpression(element.ShowCondition.Condition)
if err != nil {
        return 0, e.SMSInvalidConditionError
}
```

Any ideas?

**Comment by ralekseenkov** [comment]
I have exactly the same problem. Seems like accessors are broken.

E.g. create expression "xyz.Name == 'test'" and it will fail with
Unable to parse numeric value '.' to float64"

This will compile, but won't evaluate correctly:
"[xyz.Name] == 'test'"

**Comment by ralekseenkov** [comment]
@kennethkoontz looks like this is not implemented in 3.0.0 release (Apr 2017), but is fixed is master.

**Comment by Knetic** [comment]
The original problem is not supported yet, the initial implementation of accessors simply isn't smart enough to know that you can access the results of a function call. So something like `"Survey.Element('q1').Row('r2').Selected()"` will, unfortunately, not work (yet). 

It's a completely valid use case, and the docs do not make it clear that this wouldn't work, so I'm considering this a bug. Bad news is that i don't have a timeline for when I can get to it, and there's no easy workaround.

However the second case _should_ work, cases like `"xyz.Name == 'test'"` should be even covered in [some of the unit tests](https://github.com/Knetic/govaluate/blob/31678da578da40773346157387ef9933d650eb23/evaluation_test.go#L1275), and i can't repro the problem off master.

**Comment by sambonfire** [comment]
Do you have ideas about how you'd like this implemented?  I'd really like to see this functionality added to the package, but if I spend time implementing this I'd like to make sure the approach is something you'd approve.

**Comment by adamwasila** [comment]
@NathanZarro using tagged version with dep isn't mandatory. You can always use specific revision or branch (master) in constraints section of Gopkg.toml.
Isn't best solution but still better than manual copying while awaiting for new release of 3.x.

**Comment by javasgl** [comment]
if u use dep, change revision to `9aa49832a739dcd78a5542ff189fb82c3e423116` can resolve this problem

**Comment by wuyingsong** [comment]
I also encountered the same problem. Is there has any new progress?

**Comment by kravemir** [comment]
I have encountered the same problem, bare minimum reproduction example is:

```
func Test(t *testing.T) {
	_, err := govaluate.NewEvaluableExpression("width - extraInfo.spacing")

	assert.NilError(t, err)
}
```

Adding `govaluate` to project was evaluated as dependency in `go.mod` as:

```
	github.com/Knetic/govaluate v3.0.0+incompatible
```

And, in `go.sum` is following:

```
github.com/Knetic/govaluate v3.0.0+incompatible h1:7o6+MAPhYTCF0+fdvoz1xDedhRb4f6s9Tn1Tt7/WTEg=
github.com/Knetic/govaluate v3.0.0+incompatible/go.mod h1:r7JcOSlj0wfOMncg0iLm8Leh48TZaKVeNIfJntJ2wa0=
```

**Comment by kravemir** [comment]
Looking, that latest commit was made on Oct 22, 2017, and release on May 1, 2017. This library seems to be unsupported.

I've found alternative https://github.com/antonmedv/expr, which seems to be the best performing https://github.com/antonmedv/golang-expression-evaluation-comparison.

**Comment by litao09h** [comment]
> https://github.com/antonmedv/expr

Thanks for sharing~

**Comment by litao09h** [comment]
Encountered the same problem, 2021-11-03

**Comment by veith** [comment]
Is there any workaround for this problem?

**Comment by veith** [comment]
Not so beautiful, but this seems to work:

```go
import "github.com/iv-p/mapaccess"

var functions = map[string]govaluate.ExpressionFunction{
	"GetStringFromMap": func(args ...interface{}) (interface{}, error) {
		return mapaccess.Get(args[0], args[1].(string))
	},
}

In your evaluation you can use "GetStringFromMap(RootItem, "foo.bar")

**Comment by jtomasevic** [comment]
Hi guys, 
Any news on this? I recently install this package and having asme issue. 
Thanks,
Jovan

**Comment by windowshopr** [comment]
Yeah, would be cool to get the accessors working! Same issue here.
